python/across_api/base/schema.py

```diff
@@ -36,6 +122,68 @@ class BaseSchema(BaseModel):
     model_config = ConfigDict(from_attributes=True, arbitrary_types_allowed=True)
 
 
+class DateRangeSchema(BaseSchema):
+    """
+    Schema that defines date range
+
+    Parameters
+    ----------
+    begin
+        The start date of the range.
+    end
+        The end date of the range.
+
+    Returns
+    -------
+    data
+        The validated data with converted dates.
+
+    Raises
+    ------
+    AssertionError
+        If the end date is before the begin date.
+    """
+
+    begin: AstropyTime
+    end: AstropyTime
+
+    @model_validator(mode="after")
+    @classmethod
+    def check_dates(cls, data: Any) -> Any:
+        assert data.begin <= data.end, "End date should not be before begin"
+        return data
+
+
+class EphemSchema(BaseSchema):
+    """
+    Schema for ephemeris data.
+    """
+
+    timestamp: AstropyTimeList
+    posvec: AstropyPositionVector
+    earthsize: AstropyDegrees
+    velvec: AstropyVelocityVector
+    sun: AstropyPositionVector
+    moon: AstropyPositionVector
+    latitude: AstropyDegrees
+    longitude: AstropyDegrees
+    stepsize: AstropySeconds
+
+
+class EphemGetSchema(DateRangeSchema):
+    """Schema to define required parameters for a GET
+
+    Parameters
+    ----------
+    stepsize
+        The step size in seconds
+
+    """
+
+    stepsize: AstropySeconds
+    ...
```

Comment:
We could safely remove these since they aren't exposed by the REST API.

Reply:
Causes the same problem as above, breaking working code leading to a required refactor, but OK whatever, I removed a bunch of API code in 1ce600b and b9d08d2.